tools/src/FileContents.cc

```diff
@@ -48,11 +49,31 @@ void printContents(const char* filename, const orc::ReaderOptions opts) {
 
 int main(int argc, char* argv[]) {
   if (argc < 2) {
-    std::cout << "Usage: file-contents <filename>\n";
+    std::cout << "Usage: file-contents <filename> "
+        << "[--columns=column1,column2,...]\n" ;
     return 1;
   }
   try {
+    const std::string COLUMNS_PREFIX = "--columns=";
+    std::list<int64_t> cols;
+
+    // Read command-line options
+    char *param, *value;
+    for (int i = 2; i < argc; i++) {
+      if ( (param = std::strstr(argv[i], COLUMNS_PREFIX.c_str())) ) {
```

Comment:
It would be much more user friendly to select by column name rather than column id, which given complex types are hard to know. At that point, you might start with only selecting top level columns with something like "--columns=field1,field12", which would mean all of the types under those types.

Eventually, it would be nice to support virtual column names like "length" and "value" for lists, and "length, "key", and "value" for maps. Nested structures would look like "outer12.inner3" or "outer12.key".